build/tsc.gni

```diff
@@ -0,0 +1,40 @@
+import("npm.gni")
+
+template("typescript_build") {
+  assert(defined(invoker.tsconfig),
+         "Need tsconfig name to run (must exist in electron root)")
+  assert(defined(invoker.sources), "Need tsc sources to run")
+  assert(defined(invoker.electron_dir),
+         "Need electron_dir to run, should be 'lib' or other top level dir")
+
+  npm_action(target_name) {
+    forward_variables_from(invoker,
+                           [
+                             "deps",
+                             "public_deps",
+                             "outputs",
+                           ])
+    script = "tsc"
+
+    sources = invoker.sources
+
+    args = [
+      "-p",
+      "tsconfig." + invoker.tsconfig + ".json",
+      "--outDir",
+      rebase_path("$target_gen_dir/tsc/electron/" + invoker.electron_dir),
```

Comment:
`electron_dir` is vague. This looks like where the outputs go, so perhaps call it `output_dir` and have the invoker specify `"$target_gen_dir/tsc/electron/whatever"`? No need to enforce a particular directory structure on all users of this template.